### DISCUSSION TITLE
Welcome to esphome_components Discussions!

### DISCUSSION THREAD

**Discussion by the78mole** [the78mole/esphome_components#20] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by emjay2k** [comment]
Is there a Howto for OTA flashing updates on esphome for this device? For it seems like I am not able to connect to Fallback Hotspot (connection usually fails, and even if it succeeds it seems to crash shortly afterwards) to do it instead.

**Reply by emjay2k** [reply]
Ok found out myself: seems like installing ESPHome is required to do this but from there on its pretty automatic:

1. Install ESPHome: https://esphome.io/guides/getting_started_command_line.html (I am not a big fan of docker, so I chose the manual installation: https://esphome.io/guides/installing_esphome.html)
2. Install ESPHome dashboard: https://esphome.io/guides/getting_started_command_line.html#bonus-esphome-dashboard
3. Launch dashboard: `esphome dashboard config/`
4. Your KM271 will be detected automatically, "adopt" it and copy the api key somewhere safe
5. It will pretty much go for an update automatically from there on

Problems:
- I ran into a python error in the process (something about read_text) and had to uninstall pathlib to get rid of that `pip uninstall pathlib`
- Flashing didn't work at first (was aborted by the device), but give it several retries and it finishes some time. If that does not work, klick on the three vertical dots at the bottom right and chose `Install->Wirelessly`. After you've seen enough of the output from the device, click `STOP` and you're done

**Reply by the78mole** [reply]
Hi emjay,

great that you found out and shared your findings.
Have fun with the board :-)

Regards,
Daniel
